### PR TITLE
CmdClearAttachments fixes for #919

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

**Comment by chrisforbes** [comment]
Is clean on the internal farm.

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -8016,18 +8016,11 @@ VKAPI_ATTR void VKAPI_CALL CmdClearAttachments(VkCommandBuffer commandBuffer, ui
...
-                    if (attachment->colorAttachment == pSD->pColorAttachments[i].attachment) {
-                        found = true;
-                        break;
-                    }
-                }
-                if (!found) {
+                if (attachment->colorAttachment >= pSD->colorAttachmentCount) {
                     skip_call |= log_msg(
-                        dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_BUFFER_EXT,
+                        dev_data->report_data, VK_DEBUG_REPORT_WARNING_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_BUFFER_EXT,
```
Comment (tobine) [inline layers/core_validation.cpp]: Should this still be an error based on this valid usage language for vkCmdClearAttachments in 17.2?

`If the aspectMask member of any given element of pAttachments contains VK_IMAGE_ASPECT_COLOR_BIT, the colorAttachment member of those elements must refer to a valid color attachment in the current subpass`

This is potentially contradictory to the statement further down:

`If colorAttachment is VK_ATTACHMENT_UNUSED or is greater than or equal to VkSubpassDescription::colorAttachmentCount, then the clear has no effect.`

However, that statement could be construed to only be when VK_IMAGE_ASPECT_COLOT_BIT is NOT set. For now I'd go with the valid usage statement as higher authority and perhaps file spec ? for clarification on the potential contradiction.

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -13069,9 +13069,8 @@ TEST_F(VkLayerTest, MissingClearAttachment) {
     TEST_DESCRIPTION("Points to a wrong colorAttachment index in a VkClearAttachment "
                      "structure passed to vkCmdClearAttachments");
     ASSERT_NO_FATAL_FAILURE(InitState());
-    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT,
-                                         "vkCmdClearAttachments() attachment index 1 not found in attachment "
-                                         "reference array of active subpass 0");
+    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_WARNING_BIT_EXT,
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: WARN vs. ERROR will depend on my other comment above.

**Comment by tobine** [comment]
Change looks good but I'd like to verify the WARN vs ERROR comment.
I filed spec issue [488](https://gitlab.khronos.org/vulkan/vulkan/issues/488) for clarity.

Can we keep it ERROR for this push and then update based on the outcome of that issue?

**Comment by chrisforbes** [comment]
It's actually really weird that the spec goes out of its way to define some invalid operations as having no effect -- Vulkan doesn't do this in many other places...

**Comment by chrisforbes** [comment]
OK, changing back to ERROR for now.

**Comment by chrisforbes** [comment]
If we have some input to the resolution of 488, it would be nice to make this just UB, and get rid of the  "has no effect" language.

**Comment by chrisforbes** [comment]
Pushed to master with warning->error change.

**Pull request closed without merging**